partcad/src/partcad/converters/stl_converter.py

```
@@ -0,0 +1,100 @@
+
+        Parameters:
+        - tolerance: Linear deflection tolerance for mesh generation (larger value = fewer triangles).
+        - angular_tolerance: Angular deflection tolerance for mesh generation.
+        - ascii: Boolean flag to determine if STL should be in ASCII format.
+        """
+        logging.info(f"Attempting to write STL file: {self.output_file}")
+
+        # Check if the shape is valid
+        analyzer = BRepCheck_Analyzer(shape)
+        if not analyzer.IsValid():
+            raise RuntimeError(f"The shape is not valid and cannot be exported to STL: {shape}")
```

Ruff (0.8.2)

Avoid specifying long messages outside the exception class

(TRY003)